src/Hazelcast.Net/Clustering/ClusterSubscription.cs

```diff
@@ -132,41 +132,47 @@ public void Deactivate()
         public DateTime DeactivateTime { get; private set; }
 
         /// <summary>
-        /// Tries to add a client subscription.
+        /// Reads a subscription response.
         /// </summary>
-        /// <param name="message">The subscription response message.</param>
-        /// <param name="client">The client.</param>
-        /// <returns>Whether the client subscription was added, and its server identifier.</returns>
-        public (bool, Guid) TryAddClientSubscription(ClientMessage message, MemberConnection client)
+        /// <param name="message">The response message.</param>
+        /// <param name="connection">The connection.</param>
+        /// <returns>The corresponding member subscription.</returns>
+        public MemberSubscription ReadSubscriptionResponse(ClientMessage message, MemberConnection connection)
         {
             var serverSubscriptionId = _subscribeResponseReader(message, State);
+            return new MemberSubscription(this, serverSubscriptionId, message.CorrelationId, connection);
+        }
 
-            bool active;
-            lock (_activeLock)
+        /// <summary>
+        /// Tries to add a member subscription.
+        /// </summary>
+        /// <param name="subscription">The member subscription.</param>
+        /// <returns>Whether the member subscription was added.</returns>
+        public bool TryAddMemberSubscription(MemberSubscription subscription)
+        {
+            lock (_mutex)
             {
-                active = _active;
-                if (active)
-                    _clientSubscriptions[client] = new MemberSubscription(this, serverSubscriptionId, message.CorrelationId, client);
+                if (_active)
+                    _memberSubscriptions[subscription.Connection] = subscription;
+                return _active;
             }
-
-            return (active, serverSubscriptionId);
         }
 
         /// <summary>
         /// Removes a client subscription.
         /// </summary>
-        /// <param name="client">The client.</param>
+        /// <param name="connection">The client.</param>
```

Comment:
```suggestion
        /// <param name="connection">The connection.</param>
```

Reply:
Yes, need to rename. Mentions of "client" are old code, fixing.